### DISCUSSION TITLE
Current Day High Temp Missing on Daily Forecast

### DISCUSSION THREAD

**Discussion by hubmunkee** [jaydeethree/Home-Assistant-weatherdotcom#37] [Q&A]
Just starting to experiment with this integration, as I was not satisfied with the temp/forecast accuracy of Met.no or Pirate Weather. Great work! But I'm experiencing one issue: on the Daily Forecast, the Hi temp is missing, as well as the icon is not as expected (showing night vs day icon, when it's currently mid-afternoon). The Hourly Forecast shows as expected. Is there a setting I'm missing?

<img width="648" alt="Screenshot 2024-08-20 at 2 15 33 PM" src="https://github.com/user-attachments/assets/d1fd9cc1-fb6b-4088-b745-da630d89d0c7">

<img width="649" alt="Screenshot 2024-08-20 at 2 15 39 PM" src="https://github.com/user-attachments/assets/0f33beeb-51e9-4bb6-8c81-5c2bb5f3b4ce">

**Comment by jaydeethree** [comment]
Unfortunately this is expected behavior from Weather.com - from [their API docs](https://docs.google.com/document/d/1RY44O8ujbIA_tjlC4vYKHKzwSwEmNxuGw5sEJ9dYjG4/edit):

>the temperatureMax field in the top-level object will appear as null in the API after 3:00pm Local Apparent Time
> [...]
>The daily forecast response will transition at 3:00am Local Apparent Time.  At this time, the “0th index” day of the week will transition to the current date

So after 3:00PM local time the high temperature will disappear for the rest of the day, and it won't come back until 3:00AM the next day. Personally I dislike this behavior from their API, but there's not much I can do about it. You could create a helper entity to store the forecasted max temp and write an automation to set it each day shortly before 3:00PM, but that's the only workaround I can think of.

**Comment by hubmunkee** [comment]
I thought I had seen something written about the behavior previously, but couldn’t find or recall, so thank you for the explanation. I think I’ll try to formulate a helper to store the daily high temp and also probably just resort to using the icon from hourly forecast rather than daily, as that would be more accurate anyway. I’ll report back in case it helps anyone else. Thank you for the guidance!

Philip
